pulp_python/app/tasks/sync.py

```diff
@@ -49,239 +49,160 @@ def sync(remote_pk, repository_pk, mirror):
         raise serializers.ValidationError(
             detail=_("A remote must have a url attribute to sync."))
 
-    first_stage = PythonFirstStage(remote)
-    DeclarativeVersion(first_stage, repository, mirror).create()
-
-
-class PythonFirstStage(Stage):
-    """
-    First stage of the Asyncio Stage Pipeline.
-
-    Create a :class:`~pulpcore.plugin.stages.DeclarativeContent` object for each content unit
-    that should exist in the new :class:`~pulpcore.plugin.models.RepositoryVersion`.
-    """
-
-    def __init__(self, remote):
-        """
-        The first stage of a pulp_python sync pipeline.
-
-        Args:
-            remote (PythonRemote): The remote data to be used when syncing
-
-        """
+    with NamedTemporaryFile() as config:
+        config.write(formatted_config(remote))
+        config.seek(0)
+        first_stage = PythonBanderStage(remote, config)
+        DeclarativeVersion(first_stage, repository, mirror).create()
+
+
+def formatted_config(remote):
+    enables = "blocklist_release\n\t"
+    plugins = ""
+    if remote.includes:
+        enables += "allowlist_release\n\tallowlist_project\n\t"
+        plugins += "[allowlist]\npackages=\n\t{}".format("\n\t".join(remote.includes))
+    if remote.excludes:
+        enables += "blocklist_project\n\t"
+        plugins += "\n[blocklist]\npackages=\n\t{}".format("\n\t".join(remote.excludes))
+    if not remote.prereleases:
+        enables += "prerelease_release\n"
+
+    config_template = """
+[mirror]
+directory = /srv/pypi
+json = false
+cleanup = false
+master = {url}
+timeout = 10
+global-timeout = 1800
+workers = {concurrency}
+hash-index = false
+stop-on-error = false
+storage-backend = filesystem
+; Advanced logging configuration. Uncomment and set to the location of a
+; python logging format logging config file.  #maybe this is useful for us?
+; log-config = /etc/bandersnatch-log.conf
+verifiers = 3 
+[plugins]
+enabled = 
+    {plugin_enables}
+{plugins}
+    """.format(url=remote.url, concurrency=remote.download_concurrency, plugin_enables=enables, plugins=plugins)
+    return str.encode(config_template)
+
+
+class PythonBanderStage(Stage):
+
+    def __init__(self, remote, config):
         super().__init__()
         self.remote = remote
+        # need to take the fields stored in remote and create bander objects from it
+        Singleton._instances = {}
```

Comment:
Is this magic global state?  If so, at minimum we should document it a little better, and preferably try to kill it entirely if possible.